slidechain/slidechain_test.go

```diff
@@ -564,18 +576,24 @@ func isPostPegOutTx(tx *bc.Tx, asset xdr.Asset, amount int64, tempAddr, exporter
 	if err != nil {
 		return false
 	}
-	ref := pegOut{
-		AssetXDR: assetXDR,
-		TempAddr: tempAddr,
-		Seqnum:   seqnum,
-		Exporter: exporter,
-		Amount:   amount,
-		Anchor:   anchor,
-		Pubkey:   pubkey,
-	}
-	refdata, err := json.Marshal(ref)
-	if !bytes.Equal(refdata, tx.Log[2][2].(txvm.Bytes)) {
-		return false
+	if foundRetire {
+		ref := pegOut{
+			AssetXDR: assetXDR,
+			TempAddr: tempAddr,
+			Seqnum:   seqnum,
+			Exporter: exporter,
+			Amount:   amount,
+			Anchor:   anchor,
+			Pubkey:   pubkey,
+		}
+		refdata, err := json.Marshal(ref)
+		if err != nil {
+			log.Printf("unmarshaling json: %s", err)
+			return false
+		}
+		if !bytes.Equal(refdata, tx.Log[2][2].(txvm.Bytes)) {
```

Comment:
This is once again making the same mistake. Instead of marshaling `ref` and comparing it bytewise to the log entry, you should be unmarshaling the log entry into a separate `pegOut` and comparing that with `ref` (e.g. using `reflect.DeepEqual`).

It doesn't matter as much here, since this is just a test, and the same implementation of `json.Marshal` will be used for both this call and the call that produces the log item. The chances of those two marshalings diverging, bytewise, is practically nil. But considering that this is the PR intending to avoid that very possibility, might as well avoid it.

Reply:
I was thinking exactly what you said in the second paragraph, but it makes total sense to just avoid this possibility in the test as well, as that is the intention of the PR.